components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/PasswordGrantHandler.java

```diff
@@ -406,6 +416,7 @@ private AuthenticatedUser validateUserCredentials(OAuth2AccessTokenReqDTO tokenR
             if (StringUtils.isNotBlank(e.getErrorCode())) {
                 message = e.getErrorCode() + " " + e.getMessage();
             }
+            message = isShowAuthFailureReason ? message : "Authentication failed for " + genericErrorUserName;
```

Comment:
Don't we need this config for other grant type like code or refreshtoken?

Reply:
We would not need this for refresh token since, there's no user authentication involved.

As for the Authorization Code grant, the authentication is browser based, and showing authentication failure reasons can be handled with each authenticators.

Password grant requires this configuration since it exposes account status in the error message.